src/controllers/SafeNavigationBaseController.ts

```diff
@@ -17,14 +18,14 @@ export type RequestWithNavigation = Request & { navigation: Navigation; };
 @provide(Processor)
 class Processor implements FormActionProcessor {
     process(request: RequestWithNavigation): void {
-        const session = request.session.unsafeCoerce();
-        const applicationData: ApplicationData = session.getExtraData()
-            .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
-            .orDefaultLazy(() => {
-                const value = {} as ApplicationData;
-                session.saveExtraData(APPLICATION_DATA_KEY, value);
-                return value;
-            });
+        const session = request.session;
+
+        let applicationData: ApplicationData | undefined = session!.getExtraData(APPLICATION_DATA_KEY);
+
+        if (!applicationData) {
+            applicationData = {} as ApplicationData;
+            session!.setExtraData(APPLICATION_DATA_KEY, applicationData);
+        }
 
```

Comment:
this could be simplified
```suggestion
                    session.setExtraData(APPLICATION_DATA_KEY, applicationData || {} as ApplicationData);
```

Reply:
We dont need to set ExtraData if applicationData is defined